### PR TITLE
RF05-TC-ChristianTola-002

### PR DESCRIPTION
**Pull request by ctola91**
Test Case
image

### CODE REVIEW

**Comment by ctola91** [comment]
image

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+public class CreateIncidentPage extends BasePage {
+
+    @FindBy(xpath = "//h1[contains(text(),'Crear Incidente')]")
+    private WebElement title;
+
+    @FindBy(xpath = "//mat-radio-group/mat-radio-button/label")
+    private List<WebElement> radioGroupSeverity;
+
+    @FindBy(name = "code")
+    private WebElement codeInput;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:21]: please change this to InputFileCode

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    private WebElement checkReincident;
+
+    @FindBy(css = "#mat-checkbox-2 label")
+    private WebElement checkTreatment;
+
+    @FindBy(id = "mat-input-3")
+    private WebElement descriptionTextArea;
+
+    @FindBy(xpath = "//*[@id=\"form\"]/mat-card-actions/button[2]")
+    private WebElement saveButton;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:80]: please change to buttonSave

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    public CreateIncidentPage() {
+        CommonEvents.isVisible(saveButton);
+    }
+
+    public boolean isTitleVisible() {
+        return CommonEvents.isVisible(title);
+    }
+
+    public void setIncidentForm(List<Map<String, String>> data) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:93]: please change to setFormIncident

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        fillSelectIncidentType(data.get(0).get("IncidentType"));
+        fillSelectState(data.get(0).get("State"));
+        fillSelectArea(data.get(0).get("Area"));
+        fillReportedBy(data.get(0).get("ReportedBy"));
+        fillCheckReincident(data.get(0).get("Reincident"));
+        fillCheckTreatment(data.get(0).get("Treatment"));
+        fillDescription(data.get(0).get("Description"));
+    }
+
+    public void fillSeverityRadioBtn(String selectedRadioSeverity) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:105]: please change to fillRadioButtonSeverity

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        WebElement radioSelected = CommonEvents.findWebElement( radioGroupSeverity, selectedRadioSeverity);
+        CommonEvents.clickButton(radioSelected);
+    }
+
+    public void fillCodeInput(String code) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:110]: please change to fillInputCode

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        CommonEvents.setInputField(codeInput, code);
+    }
+
+    public void fillSelectIncidentType(String incidentTypeSelected) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:114]: I think you don't filling, you will select an option, please change to SelectedOptionIncidentType

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        CommonEvents.clickButton(selectIncidentType);
+        WebElement optionSelected = CommonEvents.findWebElement( selectIncidentTypeOptions, incidentTypeSelected);
+        CommonEvents.clickButton(optionSelected);
+    }
+
+    public void fillSelectState(String stateSelected) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:120]: I think you don't filling, you will select an option, please change to SelectedOptionState

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        CommonEvents.clickButton(selectState);
+        WebElement optionSelected = CommonEvents.findWebElement( selectStateOptions, stateSelected);
+        CommonEvents.clickButton(optionSelected);
+    }
+
+    public void fillSelectArea(String areaSelected) {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Incident/CreateIncidentPage.java:126]: I think you don't filling, you will select an option, please change to SelectedOptionArea

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 9]
```diff
@@ -110,4 +117,32 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
     public void validarListaDeEquipamientos() throws Throwable {
         Assert.assertTrue(listEquipment.isEquipmentListVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipments title"));
     }
+
+    @And("^hacemos 'click' en el boton 'Incidentes' del 'Menu Principal'$")
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:121]: please change to "seleccionamos el menu 'Incientes' del 'Menu principal'"

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 10]
```diff
@@ -110,4 +117,32 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
...
+    public void hacemosClickEnElBotonIncidentesDelMenuPrincipal() throws Throwable {
+        createIncidentPage = incidentPage.clickOnAddNewIncidentBtn();
+    }
+
+    @When("^Llenamos los 'datos del formulario' dentro del formulario de 'Creacion de Incidentes'$")
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:126]: please change to "insertamos informacion valida en el formulario de 'Creacion de Incidentes'

File: src/test/resources/features/Incident/Incident.feature [hunk 11]
```diff
@@ -5,7 +5,17 @@ Feature: Lista de Incidentes
...
+
+  Scenario: RF05-TC-002 - Verificar que el sistema pueda crear incidentes correctamente.
+    Given 'Menu principal' estee cargado
+    And Presionar en la opcion 'Incidentes' del 'Menu Principal'
+    And Verificar que la tabla de incidentes se muestre correctamente
+    And hacemos 'click' en el boton 'Incidentes' del 'Menu Principal'
+    When Llenamos los 'datos del formulario' dentro del formulario de 'Creacion de Incidentes'
+    | Severity | Code   | IncidentType | State     | Area           | ReportedBy | Reincident | Treatment | Description           |
+    | Media    | ACC-02 | enfermedad   | entregado | Remodelaciones   | juan       | true       | true     | this is a description |
+    And Hacemos clic en el boton de 'Guardar'
```
Comment (marbusjim) [inline src/test/resources/features/Incident/Incident.feature:21]: What is your assertion?

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 12]
```diff
@@ -110,4 +117,32 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
...
+        createIncidentPage.clickSaveButton();
+        Assert.assertTrue(createIncidentPage.isVisibleToastMessage(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Toast Message Success"));
+    }
+
+    @And("^Verificamos que el titulo de 'Creacion de Incidentes' se muestre correctamente$")
+    public void verificamosQueElTituloDeCreacionDeIncidentesSeMuestreCorrectamente() throws Throwable {
+        Assert.assertTrue(createIncidentPage.isTitleVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Titulo 'Crear Incidente'"));
+    }
+
+    @When("^Llenamos el campo 'code' con un valor vacio$")
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:143]: I don't see where you use this

**Review by marbusjim — COMMENTED** [review]
please review the comments

**Pull request closed without merging**